exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/conv/DateEpochBEConvertFrom.java

```diff
@@ -41,6 +41,7 @@ public void eval() {
 
     in.buffer.readerIndex(in.start);
     long epochMillis = Long.reverseBytes(in.buffer.readLong());
+    in.buffer.readerIndex(0);
```

Comment:
Can you explain why this is necessary? It seems like the issue you are trying to resolve is related to whatever comes after this function rather than an issue with this function. There should not guarantees about the reader index of a Drillbuf (in fact, we should remove this from being exposed in functions). See how this function positions the reader before doing anything else. Is there some other operation which doesn't position the reader?